app/Observers/CampaignObserver.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace App\Observers;
+
+use App\Models\Campaign;
+
+class CampaignObserver
+{
+    /**
+     * Handle the Campaign "created" event.
+     *
+     * @param  \App\Models\Campaign  $campaign
+     * @return void
+     */
+    public function created(Campaign $campaign)
+    {
+        $campaign->code = $campaign->generateCode();
+        $campaign->save();
+    }
```

Comment:
technically you should be able to do 
```suggestion
    public function creating(Campaign $campaign)
    {
        $campaign->code = $campaign->generateCode();
    }
```

Saves you a database query; either way it works though